xmtp_mls/src/groups/members.rs

```diff
@@ -0,0 +1,90 @@
+use std::collections::HashMap;
+
+use xmtp_proto::api_client::{XmtpApiClient, XmtpMlsClient};
+
+use crate::identity::Identity;
+
+use super::{GroupError, MlsGroup};
+
+#[derive(Debug, Clone)]
+pub struct GroupMember {
+    pub wallet_address: String,
+    pub installation_ids: Vec<Vec<u8>>,
+}
+
+impl<'c, ApiClient> MlsGroup<'c, ApiClient>
+where
+    ApiClient: XmtpApiClient + XmtpMlsClient,
+{
+    // Load the member list for the group from the DB, merging together multiple installations into a single entry
+    pub fn members(&self) -> Result<Vec<GroupMember>, GroupError> {
+        let openmls_group =
+            self.load_mls_group(&self.client.mls_provider(&mut self.client.store.conn()?))?;
+
+        let member_map: HashMap<String, GroupMember> = openmls_group
+            .members()
+            .filter_map(|member| {
+                Identity::get_validated_account_address(
+                    member.credential.identity(),
+                    &member.signature_key,
+                )
+                .ok()
```

Comment:
Any reason we are filtering this? Doesn't this effectively make illegitimate group members invisible? Wondering if we should at least log something

Reply:
I can add some logging. I would expect we'll want to do the checks for legitimacy before actually accepting the commit to let them into the group, so illegitimate users should be impossible. But it would be good to have some visibility if we somehow screw that up.

Reply:
Or like, just fail the group completely, seeing as it has been compromised. Your call